src/backend/catalog/heap.c

```diff
@@ -1070,6 +1071,15 @@ AddNewAttributeTuples(Oid new_rel_oid,
 				SysAtt[i]->attnum == ObjectIdAttributeNumber)
 				continue;
 
+			/* skip xmin, xmax, cmin, cmax for AO table */
+			if (appendOnlyRel &&
+				(SysAtt[i]->attnum == MinTransactionIdAttributeNumber ||
+				 SysAtt[i]->attnum == MinCommandIdAttributeNumber ||
+				 SysAtt[i]->attnum == MaxTransactionIdAttributeNumber ||
+				 SysAtt[i]->attnum == MaxCommandIdAttributeNumber
+				 ))
+				continue;
```

Comment:
Do we still need do not construct these columns for AO-tables?

Reply:
This question can be reformulated: Why do we need to have these for AO tables if that columns are not exists and always report error. I don't see any reason to have them, so suggest to remove.

Reply:
7.x does not keep xmin, xmax, cmin, cmax in the pg_attribute, similar to this solution. I believe we may remove this code from PR to avoid potential problems with backward compatibility with existing databases.

Reply:
7.x does not keep xmin, xmax, cmin, cmax in the pg_attribute, similar to this solution. I believe we may remove this code from PR to avoid potential problems with backward compatibility with existing databases.

Reply:
> we may remove this code from PR

let so do it